src/utils/custom-highlight-lang/swift.js

```diff
@@ -41,5 +41,37 @@ export default function swiftOverride(hljs) {
     };
   }
 
+  // Checks if a given language sub-mode matches the "ESCAPED_NEWLINE" from the
+  // built-in Swift parser from hljs
+  const isEscapedNewlineMode = (mode) => {
+    const { className, match } = mode;
+    if (className !== 'subst' || !match) {
+      return false;
+    }
+
+    const matchStr = match.toString();
+    return matchStr.startsWith('\\') && matchStr.endsWith('[\\t ]*(?:[\\r\\n]|\\r\\n)');
+  };
+
+  // replace the "ESCAPED_NEWLINE" sub-mode in the multiline string literal mode
+  // variants so that it doesn't include the actual newline characters in the
+  // span token that it generates, because this causes issues with our
+  // line-number + multi-line string literal logic when the span for the
+  // backslash token is split across multiple lines
+  const str = language.contains.find(({ className }) => className === 'string');
+  str.variants = str.variants.map(variant => ({
```

Comment:
This is relying changing the `variants` inside a single object in the `contains` array, purely by reference? 🤣 pure 🪄 🦄

Reply:
Yeah, I don't love it tbh. Maybe I could refactor it to not rely on changing it by reference, but I worry about how much more code I need to write 😿

Reply:
Here's a less 🪄🦄 version: eaa410127784b057baa8b03db9287a5ab172d2bf 😆